src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/HtmlHelper.cs

```diff
@@ -522,6 +523,23 @@ public Task RenderPartialAsync(string partialViewName, object model, ViewDataDic
             string templateName,
             object additionalViewData)
         {
+            if (modelExplorer.Metadata.IsEnum && modelExplorer.Model != null)
+            {
+                var value = (modelExplorer.Model as Enum).ToString("d");
```

Comment:
This will hit an NRE if `Model` is `null` which can happen because `ModelExplorer` doesn't enforce that `Model` is compatible with `ModelType` e.g. `string`s are sometimes used. `Model` can also be `null` and compatible w/ `ModelType` i.e. when `ModelType` is a `Nullable<TEnum>`. Suggest saving the `as Model` result in a variable then `null`-checking that variable in the condition just above.

General point is to never dereference a conditional cast without handling `null`.

Think I missed this in the last iteration, sorry.